### PR TITLE
chore: integrate service-account SDK

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Integrate `serviceaccount-sdk` with the RHOAS CLI

### Verification Steps
1. Build the CLI from the changes and try out the service account commands.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [X] Dependency update

### ToDo

- [X] Format the time in `service-account list`
- ~~[ ] Distinguish between field "name" and "description" in `ServiceAccountData` object~~

### CODE REVIEW

**Comment by wtrocki** [comment]
Looks good for me. No issues

**Comment by rkpattnaik780** [comment]
Hi @wtrocki ,
[Unrelated to PR] Noticed that staging environment in both CLI and UI use production auth url for service account operations. Has there been any discussion regarding use of the staging sso url: https://github.com/redhat-developer/app-services-sdk-go/blob/7112e88954ab96caf33a3550953a56e9f89fcf19/.openapi/service-accounts.yaml#L11

File: pkg/shared/connection/api/defaultapi/default_client.go [hunk 1]
```diff
@@ -73,16 +76,16 @@ func (a *defaultAPI) ServiceRegistryMgmt() registrymgmtclient.RegistriesApi {
 }
 
 // ServiceAccountMgmt return a new Service Account Management API client instance
-func (a *defaultAPI) ServiceAccountMgmt() kafkamgmtclient.SecurityApi {
+func (a *defaultAPI) ServiceAccountMgmt() svcacctmgmtclient.ServiceAccountsApi {
 	tc := a.CreateOAuthTransport(a.AccessToken)
-	client := kafkamgmt.NewAPIClient(&kafkamgmt.Config{
-		BaseURL:    a.ApiURL.String(),
+	client := svcacctmgmt.NewAPIClient(&svcacctmgmt.Config{
+		BaseURL:    build.ProductionAuthURL,
```
Comment (wtrocki) [inline pkg/shared/connection/api/defaultapi/default_client.go:82]: I do not think this is right. I think we should connect it with the login flag we have and config. Build is compile time value

Comment (wtrocki) [reply pkg/shared/connection/api/defaultapi/default_client.go:82]: @mikeedgar @andreaTP @riccardo-forina   for there will be some extra implications for UI, kas-instaler and srs-installer here.
Any other service that uses custom keycloak (not sso.redhat.com) will stop working when we merge that PR.
We were asked to migrate from kafka control plane service accounts  API to use Service Accounts provided by CIAM. This API has different url (not api.openshift.com). Service is hosted on gitlab - not sure if we can deploy this with kas-installer etc. Even if we deploy this keycloak url will still be used by all clients making it hard to integrate.

Comment (machi1990) [reply pkg/shared/connection/api/defaultapi/default_client.go:82]: +1 on making this configurable either via CLI flags or it being read dynamically from the control plane sso providers endpoint. 

> Any other service that uses custom keycloak (not sso.redhat.com) will stop working when we merge that PR.

I think @akoserwal is working on this. He may be able to chime in here with details.

Comment (wtrocki) [reply pkg/shared/connection/api/defaultapi/default_client.go:82]: After initial discussion I think we good to assume the sso url as api root for this API

Comment (andreaTP) [reply pkg/shared/connection/api/defaultapi/default_client.go:82]: > +1 on making this configurable either via CLI flags or it being read dynamically from the control plane sso providers endpoint.

+1 as well!

Comment (MikeEdgar) [reply pkg/shared/connection/api/defaultapi/default_client.go:82]: > +1 on making this configurable either via CLI flags or it being read dynamically from the control plane sso providers endpoint.

Another +1. Having the base URL for a service configurable expands the ability to use services more easily outside of prod.

**Review by wtrocki — APPROVED** [review]

File: internal/build/build.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ var (
 	DefaultLoginTimeout         = 60 * time.Second
 	OfflineTokenURL             = ConsoleURL + "/openshift/token"
 	ProductionAuthURL           = "https://sso.redhat.com/auth/realms/redhat-external"
+	StagingAuthURL              = "https://sso.stage.redhat.com/auth/realms/redhat-external"
```
Comment (wtrocki) [inline internal/build/build.go:63]: +1. Thank you for adding it.

**Review by wtrocki — APPROVED** [review]

**Pull request merged**